### PR TITLE
Remove dependency to jsdom and update tests

### PR DESCRIPTION
**Pull request by gfontorbe**
Replace dependency to 'jsdom' by 'happy-dom' due to license conflicts.

Also enable tests on .tsx files.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "eslint-config-prettier": "^9.0.0",
     "eslint-plugin-header": "^3.1.1",
     "eslint-plugin-no-null": "^1.0.2",
+    "happy-dom": "^12.10.3",
```
Comment (spoenemann) [inline package.json:20]: I'd move this to packages/sprotty/package.json

**Review by spoenemann — COMMENTED** [review]
Please also remove the test-helper from the exclude list in tsconfig.json

**Comment by dhuebner** [comment]
test-helper also still mentioned in `packages/sprotty/package.json`, line 45 and 56

**Comment by spoenemann** [comment]
Those are already removed in #400.

File: packages/sprotty/package.json [hunk 2]
```diff
@@ -34,8 +34,6 @@
   "devDependencies": {
     "@types/chai": "^4.3.6",
```
Comment (dhuebner) [inline packages/sprotty/package.json:35]: @gfontorbe 
Could you also remove this dependency to `@types/chai` please? It is not needed I think.

**Review by dhuebner — APPROVED** [review]
LGTM

**Review by spoenemann — APPROVED** [review]
Thanks!

**Pull request merged**